internal/attributed-string-slice/attributed_string_slice.go

```diff
@@ -0,0 +1,79 @@
+package attributedstringslice
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+
+	"github.com/BurntSushi/toml"
+)
+
+type attributedStringSlice struct { // A "mixed-type array" in TOML.
+	slice      []string
+	attributes struct { // Using a struct allows for adding more attributes in the feature.
+		append *bool // Nil if not set by the user
+	}
+}
+
+func (ts *attributedStringSlice) UnmarshalTOML(data interface{}) error {
+	iFaceSlice, ok := data.([]interface{})
+	if !ok {
+		return fmt.Errorf("unable to cast to interface array: %v", data)
+	}
+
+	var loadedStrings []string
+	for _, x := range iFaceSlice { // Iterate over each item in the slice.
+		kind := reflect.ValueOf(x).Kind()
+		switch kind {
+		case reflect.String: // Strings are directly appended to the slice.
+			loadedStrings = append(loadedStrings, fmt.Sprintf("%v", x))
+		case reflect.Map: // The attribute struct is represented as a map.
```

Comment:
I think this is better written as 
```
switch val := x.(type) {
	case string:
		slice = append(slice, val)
	case map[string]interface{}:
		...
	default:
		// no match
	}
```

This avoids the extra type casts you need

Reply:
I did not know such a switch table is possible. Thanks for sharing!